src/Omnipay/Common/Message/AbstractResponse.php

```diff
@@ -182,8 +183,9 @@ public function getRedirectResponse()
             throw new RuntimeException('This response does not support redirection.');
         }
 
+        /** @var $this RedirectResponseInterface */
         if ('GET' === $this->getRedirectMethod()) {
-            return HttpRedirectResponse::create($this->getRedirectUrl());
+            return new RedirectResponse($this->getRedirectUrl());
```

Comment:
@barryvdh yes you are right about that. Header location + status , what I meant is use $response->withHeader('Location', $this->getRedirectUrl()) , instead of calling RedirectResponse . Regarding what I mean hard is for I don't know omnipays internal and not sure if that is hard to implement for it needs the $response to be passed for it is immutable.

@judgej yes, I hope I am well aware of it :-) . I am one of the early contributors to the project.